### DISCUSSION TITLE
SSL certificate

### DISCUSSION THREAD

**Discussion by roguefaction** [abstracta/jmeter-java-dsl#63] [Q&A] [answered]
Hi, 

is there any way to add an SSL truststore/keystore to HTTP requests in a test plan?

Vytautas

**Comment by rabelenda** [comment, marked as answer]
Hello,

Thank you for asking this question, is a very important one in some scenarios. 

Have you tried setting `javax.net.ssl.keyStore` and `javax.net.ssl.keyStorePassword` java system properties? 

Eg:

```java
System.setProperty("javax.net.ssl.keyStore", "myKeystore.jks");
System.setProperty("javax.net.ssl.keyStorePassword", "myPass");
```

In general you wouldn't need to define a truststore since jmeter by defaults trusts all keys. If you need you might try with `javax.net.ssl.trustStore` system property.

If setting properties programmatically (with `System.setProperty`) you might also try setting it in the JVM that runs the tests, for example, if you use maven:

```xml
<build>
  <plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-surefire-plugin</artifactId>
      <version>3.0.0-M5</version>
      <configuration>
        <argLine>-Djavax.net.ssl.keyStore=myKeystore.jks -Djavax.net.ssl.keyStorePassword=myPass</argLine>
      </configuration>
    </plugin>
  </plugins>
</build>
```

Regards

**Reply by roguefaction** [reply]
Setting system variables worked. Thank you!
